### ISSUE TITLE
Show revision in unstable/development svn builds

### ISSUE THREAD

**Issue by aircrack-ng** [aircrack-ng/aircrack-ng#39]
**Reported by hirte on 3 Jan 2007 21:36 UTC**

Define 
```
-D_DEVEL="`cat .svn/entries | grep revision | sed 's/.*revision=\"\(.*\)\".*/\1/'`"
```

as an OPTFLAGS option in the Makefile and delete "#define _DEVEL 0" in version.h

'''We have to reverse that for the stable versions! '''

But using that method gives us way better control of hunting down bugs in svn revisions, as users can post us the exact version from the command they used (<command> | grep dev) , instead of trusting their intuition for using "the newest checkout" ...

Personally i don't like this rather complex command to be used as an option for gcc, because of the high failure potential it involves. So I'd like to read some alternatives as a response to this ticket ;).

**Comment by aircrack-ng** [comment]
**Comment by hirte on 3 Jan 2007 21:36 UTC**

(In #67) Added revision definition in Makefile (Closes #39).

**Comment by aircrack-ng** [comment]
**Comment by anonymous on 3 Jan 2007 21:36 UTC**

Revision 72 Build Error in aircrack-ng.c

```
gcc -g -W -Wall -O3 -D_FILE_OFFSET_BITS=64 -D_REVISION="`cat .svn/entries | grep revision | sed 's/.*revision=\"\(.*\)\".*/\1/'`" src/aircrack-ng.c src/crypto.c src/sha1-mmx.S src/common.c -o aircrack-ng -lpthread
src/aircrack-ng.c: In Funktion main:
src/aircrack-ng.c:2779: Fehler: expected expression before ) token
make: *** [aircrack-ng] Fehler 1
```

**Comment by aircrack-ng** [comment]
**Comment by hirte on 3 Jan 2007 21:36 UTC**

That's what I was worried about, maybe removing the quotation solves this... aircrack-ng/aircrack-ng@66c4548c29686f5e47826daf39cb83e244dc4787

```
-D_REVISION=`svn info | grep Revision | cut -b 11-12`
```
could also work.

**Comment by aircrack-ng** [comment]
**Comment by gandalf on 3 Jan 2007 21:36 UTC**

(In #74) Fix svn revision detection in Makefile and make it works it's we're not in a rev copy (ie for release) (Closes: #39).
Make this flags independent from CFLAGS/OPTFLAGS as it shouldn't be modified.
